parsers/mysql/mysql.go

```diff
@@ -331,15 +336,32 @@ func (p *Parser) ProcessLines(lines <-chan string, send chan<- event.Event, pref
 			if foundStatement {
 				// we've started a new event. Send the previous one.
 				foundStatement = false
-				rawEvents <- groupedLines
+				// if sampling is enabled, drop events here to conserve parsing CPU
+				if p.SampleRate > 1 {
+					if rand.Intn(p.SampleRate) == 0 {
```

Comment:
i think we don't want that, or else we'll hit line 347 if `p.SampleRate > 1` and we should drop it (aka `rand.Intn(p.SampleRate) != 0`)